sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -929,6 +930,49 @@ async def _message_generator_async(self, timeout=None):
             if self._shutdown:
                 await self.close_async()
 
+    async def _on_disposition_received_async(self, message_delivery, reason, state):
+        if reason == LinkDeliverySettleReason.DISPOSITION_RECEIVED:
+            state_error = list(state.values())[0]
+            message_delivery.reason = reason
+            if state and (len(state_error) == 0 or state_error[0] is None):
+                message_delivery.state = MessageDeliveryState.Ok
+            else:
+                try:
+                    error_info = state[SEND_DISPOSITION_REJECT]
+                    self._process_receive_error(
+                        message_delivery,
+                        condition=error_info[0][0],
+                        description=error_info[0][1],
+                        info=error_info[0][2]
+                    )
+                except TypeError:
+                    self._process_receive_error(
+                        message_delivery,
+                        condition=ErrorCondition.UnknownError
+                    )
+        elif reason == LinkDeliverySettleReason.TIMEOUT:
+            message_delivery.state = MessageDeliveryState.Timeout
```

Comment:
Better way to set up timeout?

Reply:
this raises a `ServiceBusError: Handler closed: The service did not respond in time, message may or may not have been settled.`     OperationTimeoutError is retryable which is not what we want here?

Reply:
yeah we dont want to retry here, this would be a case where the user should retry settling the message. The only time we would maybe want to retry is when the disposition doesnt come in at all or is missed. Other times the service would have a reason to not settle it and retrying would be of no use. So overall not retrying is what Im thinking